libmscore/rendermidi.cpp

```diff
@@ -1227,16 +1241,20 @@ void Score::renderStaff(EventMap* events, Staff* staff, DynamicsRenderMethod met
             for (Measure* m = tick2measure(startTick); m; m = m->nextMeasure()) {
                   if (lastMeasure && m->isRepeatMeasure(staff)) {
                         int offset = (m->tick() - lastMeasure->tick()).ticks();
-                        collectMeasureEvents(events, lastMeasure, staff, tickOffset + offset, method, cc, lastHairpinStart);
+                        collectMeasureEvents(events, lastMeasure, staff, tickOffset + offset, method, cc, lastHairpinStart, staffTempVelEvents);
                         }
                   else {
                         lastMeasure = m;
-                        collectMeasureEvents(events, lastMeasure, staff, tickOffset, method, cc, lastHairpinStart);
+                        collectMeasureEvents(events, lastMeasure, staff, tickOffset, method, cc, lastHairpinStart, staffTempVelEvents);
                         }
                   if (m->tick() + m->ticks() >= endTick)
                         break;
                   }
             }
+
+      for (std::pair<int, NPlayEvent> event : staffTempVelEvents) {
+            events->insert(std::pair<int, NPlayEvent>(event.first, event.second));
```

Comment:
I think each staff has its own channel, because I can get single note dynamics to work independently on a grand staff for something like an accordion. I'll check that, though.